File: flutter/lib/domain/use_case/deck/create_new_deck_use_case.dart
import 'package:flashcard/domain/repository/flashcard_repository.dart';

import '../../entities/deck.dart';
import '../../repository/deck_repository.dart';

class CreateNewDeckUseCase {
  final DeckRepository _deckRepository;

  CreateNewDeckUseCase(this._deckRepository);

  Future<int> call(Deck deck) async {
    return await _deckRepository.addDeck(deck);
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/CreateNewDeckUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase

import com.synergyboat.flashcardAi.domain.entities.Deck
import com.synergyboat.flashcardAi.domain.repository.DeckRepository
import javax.inject.Inject

class CreateNewDeckUseCase @Inject constructor(
    private val repository: DeckRepository
) {
    suspend operator fun invoke(deck: Deck): Deck {
        return repository.createDeck(deck)
    }
}